File: java/p006.java
/**
 * p006 - Compute the difference between the sum of the squares up 
 *        and the square of the sum of the first 100 natural numbers.
 */
public final class p006 {
  
  public static void main(String[] args) {
    System.out.println(new p006().solve());
  }

  public int solve() {
    //System.out.println(squareOfSums(100));
    //System.out.println(sumOfSquares(100));
    return squareOfSums(100) - sumOfSquares(100);
  }

  private static int sumOfSquares(int n) {
    int sum = 0;
    for (int i = 1; i <= n; i++) sum += Math.pow(i, 2);
    return sum;
  }

  private static int squareOfSums(int n) {
    int sum = 0;
    for (int i = 1; i <= n; i++) sum += i;
    return (int)Math.pow(sum, 2);
  }
}

File: src/typescript/p006.ts
/*
  PROBLEM 006
  Find the difference between the sum of the squares of the first 
  one hundred natural numbers and the square of the sum.
*/

/*
  Compute the difference between the sum of the squares up to a given number (num) and the square
  of the sum up to the number.
*/
export const sumSquareDifference = (num: number): number => {
  const sumOfSquares = (num: number): number => {
    if (num < 1) { return 0; }
    return Math.pow(num, 2) + sumOfSquares(num - 1);
  };
  const squareOfSums = (num: number): number => {
    const sum = [...Array(num).keys()].reduce((n, sum) => n + sum, 0);
    return Math.pow(sum, 2);
  };

  return squareOfSums(num) - sumOfSquares(num);
};
